.github/workflows/pr-tests-code.yml

```diff
@@ -0,0 +1,60 @@
+name: Run code tests
+
+on:
+  pull_request:
+    branches: 
+      - master
+      - develop
+    paths: 
+      - src/**
+
+jobs:
+  ubuntu:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v2
+      - name: Use .NET Core 2.1 LTS SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '2.1.x'
+      - name: Use .NET Core 3.1 LTS SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.x'
+      - name: Install dependencies
+        working-directory: src
+        run: dotnet restore
+      - name: Build solution (release config)
+        working-directory: src
+        run: dotnet build --no-restore -c Release -p:NoWarn=CS1591
```

Comment:
Why is the packages restore taken as a separate step? So is it preferable to do instead of ` run: dotnet build -c Release -p:NoWarn=CS1591`?

Reply:
Ah! Because of second build step with Debug configuration I think.

Reply:
And dotnet test also ).

Reply:
We could just write `dotnet test` and ignore all these prior steps.  But it's easier to review a workflow run when the individual tasks are separated.  I also added an extra step to build it twice - once for release and once for debug.  Debug is used for tests.  It would really only matter if there was compile-time conditionals in the code.  But as we are not billed for workflow run time (and it really doesn't take long to build), I included that step.